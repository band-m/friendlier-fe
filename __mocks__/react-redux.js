jest.mock('react-redux', () => {
  return {
    __esModule: true,
    useDispatch: jest.fn(() => () => {}),
    useSelector: jest.fn((selector) => selector({ contacts: { contactList: [] }, auth: { user: '' } }))
  };
});
