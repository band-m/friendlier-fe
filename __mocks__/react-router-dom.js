jest.mock('react-router-dom', () => {
  return {
    useHistory: jest.fn({})
  };
});
