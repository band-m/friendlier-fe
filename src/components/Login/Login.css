.Login{
  margin: 1rem;
  display:flex;
  flex-direction: column;
  align-items: center;

  h1{
    font-size: 1.4rem;
    padding: 1rem 0 1.5rem;
  }
  label{
    input[type="email"] {
      margin-left: 3rem;
    }
    input[type="password"] {
      margin-left: 1.1rem;
    }
  }
  button {
    margin: 1rem 0;
  }
}
