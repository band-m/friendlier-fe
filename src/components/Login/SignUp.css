@import url('https://fonts.googleapis.com/css?family=DM+Serif+Display|Hepta+Slab:300,400,600,900|Lato:200,300,400,700|Raleway:100,200,300,400,500,700,900|Rozha+One&display=swap');

