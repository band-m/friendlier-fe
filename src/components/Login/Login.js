import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import styles from './Login.css';
import { login } from '../../data/actions/auth-actions';
import { useDispatch } from 'react-redux';

const Login = () => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const dispatch = useDispatch();
  const handleSubmit = event => {
    event.preventDefault();
    dispatch(login(email, password));
  };

  return (
    <main className={styles.Login}>
      <h1>Login</h1>
      <form onSubmit={handleSubmit}>
        <div><label htmlFor="email">Email</label><input type="email" name="email" value={email} onChange={({ target }) => setEmail(target.value)} /></div>
        <div><label htmlFor="password">Password</label><input type="password" name="password" value={password} onChange={({ target }) => setPassword(target.value)} /></div>
        <button>Login</button>
      </form>
      <p className={styles.center}><Link to="/signup">Sign up</Link></p>
    </main>
  );
};

export default Login;
