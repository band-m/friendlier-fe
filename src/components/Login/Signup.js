import React, { useState } from 'react';
import { Link, useHistory } from 'react-router-dom';
import styles from './Login.css';
import { signup } from '../../data/actions/auth-actions';
import { useDispatch, useSelector } from 'react-redux';
import { selectUser, selectError, selectLoading } from '../../data/selectors/auth-selector';

const Signup = () => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [username, setUsername] = useState('');
  const dispatch = useDispatch();
  const history = useHistory();
  const user = useSelector(selectUser);
  const error = useSelector(selectError);
  const loading = useSelector(selectLoading);
  const handleSubmit = event => {
    event.preventDefault();
    dispatch(signup(username, email, password));
  };

  if(user){
    history.push('/');
  }

  return (
    <main className={styles.Login}>
      <h1>Signup</h1>
      {error && <p>{error.message}</p>}
      <form onSubmit={handleSubmit}>
        <div><label htmlFor="username">Username</label><input type="text" name="username" value={username} onChange={({ target }) => setUsername(target.value)} /></div>
        <div><label htmlFor="email">Email</label><input type="email" name="email" value={email} onChange={({ target }) => setEmail(target.value)} /></div>
        <div><label htmlFor="password">Password</label><input type="password" name="password" value={password} onChange={({ target }) => setPassword(target.value)} /></div>
        <button>Signup</button>
      </form>
      <p className={styles.center}><Link to="/">Login</Link></p>
      {loading && <img src='../../../public/assets/loading_spinner.gif' alt='loading-spinner' />}
    </main>
  );
};

export default Signup;
