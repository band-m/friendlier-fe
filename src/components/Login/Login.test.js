import React from 'react';
import 'react-redux';
import 'react-router-dom';
import { shallow } from 'enzyme';
import Login from './Login';

describe('Login component', () => {
  it('renders Login', () => {
    const wrapper = shallow(<Login />);
    expect(wrapper).toMatchSnapshot();
  });
});
