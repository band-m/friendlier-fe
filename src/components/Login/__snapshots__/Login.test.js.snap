// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders Login 1`] = `
<main
  className="Login"
>
  <h1>
    Login
  </h1>
  <form
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        name="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div>
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <button>
      Login
    </button>
  </form>
  <p
    className="center"
  >
    <Component
      to="/signup"
    >
      Sign up
    </Component>
  </p>
</main>
`;
